core/lib/libtomcrypt/src/headers/tomcrypt_cipher.h

```diff
@@ -466,11 +466,12 @@ extern const struct ltc_cipher_descriptor {
        @param ct      Ciphertext
        @param blocks  The number of complete blocks to process
        @param IV      The initial value (input/output)
+       @param IV_size The initial value size
```

Comment:
There's a comment in `aes_ctr_encrypt_nblocks()`:

> /* In big endian mode IV_size=0 -> 128 bits and IV_size=8 -> 64 bits */

How does that relate to this value, is the value in bytes/bits or on some other magic scale?

Reply:
Have a look at computation of CTR length for big endian case [here](https://github.com/OP-TEE/optee_os/blob/master/core/lib/libtomcrypt/src/modes/ctr/ctr_start.c#L48).

Basically for BE case CTR length is reversed meaning 0 -> 16 bytes, 1 -> 15 bytes, ..., 8 -> 8 bytes, and so on.

Reply:
For LE case, it would be normal: 0 -> 0 bytes, 1 -> 1 bytes, and so on.

Reply:
My point is that the comment "The initial value size" is not very specific. You need to search the code to get some clue and even then it's not obvious.

Reply:
How about "The initial value size identifier"?

Reply:
I think a complete description is needed.

Reply:
Will add complete description.